### ISSUE TITLE
Docker does not support parallel pulls

### ISSUE THREAD

**Issue by zopyx** [docker/docker#10119]
Running Docker 1.3.2 on latest CentOS 7.0, complete new installation

I start to parallel two Docker run command simulataneously in two terminal windows

docker run zopyx/xmldirector-plone
docker run zopyx/existdb-22

One of the commands started  to pull its layer after the first command finished completely.
Why is not possible to start to multiple docker run command in parallel? Why is there a global lock while pulling layers? Most of pulling one layer is network IO and I see zero need for keeping a global lock here while pulling all layers of one images and letting other docker command die of starvation meanwhile?!

**Comment by zopyx** [comment]
On the other hand it seems to work run a docker run and a docker build in parallel...no idea where the difference is.

**Comment by icecrime** [comment]
Hello! I believe parallel pull is part of #9784 (which *might* be in the upcoming Docker 1.5). It's something we are aware of anyways, but thanks for reporting.

**Comment by jfrazelle** [comment]
This is definitely a duplicate I'll find the other in a sec unless someone
beats me

**Comment by jfrazelle** [comment]
closing as duplicate of #7335, please place all idea etc there
